libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/GoogleFragment.java

```diff
@@ -225,6 +225,7 @@ protected void finishFlow() {
     }
 
     protected void showError(String message) {
+        mAnalyticsListener.trackFailure(message);
         finishFlow();
         mGoogleListener.onGoogleSignupError(message);
     }
```

Comment:
I was initially confused on why this wasn't being handled inside the `onGoogleSignupError` callback but after looking at `UnifiedLoginTracker.trackFailure()` I'm assuming it's to preserve the current flow and step. If that's the case, I think we need a way to communicate this behavior more clearly. `UnifiedLoginTracker` is currently not a part of the login library, so other library clients will only be able to look at `LoginAnalyticsListener.trackFailure()` to get more context on how this is supposed to work, which doesn't make it obvious. Let me know what you think.

Reply:
Another thought I just had about this: do we really need to track the current flow and step along with failures and clicks? I'm not very familiar with Tracks, but wouldn't we be able to use it to check what was the last tracked flow/step before a certain failure event was tracked?

Reply:
> I was initially confused on why this wasn't being handled inside the onGoogleSignupError callback but after looking at UnifiedLoginTracker.trackFailure() I'm assuming it's to preserve the current flow and step. If that's the case, I think we need a way to communicate this behavior more clearly. UnifiedLoginTracker is currently not a part of the login library, so other library clients will only be able to look at LoginAnalyticsListener.trackFailure() to get more context on how this is supposed to work, which doesn't make it obvious. Let me know what you think.
Hm, this is an interesting catch. Not having the `trackFailure` in the `onGoogleSignupError` is better and it should be there. However you're right that we might lose the flow/step by this change. I'll look into it. 
> Another thought I just had about this: do we really need to track the current flow and step along with failures and clicks? I'm not very familiar with Tracks, but wouldn't we be able to use it to check what was the last tracked flow/step before a certain failure event was tracked?
My thinking behind this is that it will be much easier to analyze if a certain step is failing if we have the data also in the failure event. If you don't, you have to use funnels to get that information. I've asked the data team about this and they agree that it is useful to keep the flow and the step on the failure events. We should just make sure we're tracking the correct information :-)

Reply:
I've moved the failure tracking to the `onGoogleSignupError`. I've tested it (slow data -> try google login -> turn on airplane mode -> failure + dialog) and it works correctly and keeps the last flow & the step. I think this approach is better 👍